pkg/controllers/termination/terminate.go

```diff
@@ -77,7 +81,15 @@ func (t *Terminator) drain(ctx context.Context, node *v1.Node) (bool, error) {
 		}
 		// Don't attempt to evict a pod that's already evicting
 		if !pod.DeletionTimestamp.IsZero() {
-			continue
+			// If the kubelet cannot register a pod as deleted, we need to force delete the pod in order to terminate the node
+			if time.Since(pod.DeletionTimestamp.Time) > time.Duration(*pod.DeletionGracePeriodSeconds) {
+				if err := t.KubeClient.Delete(ctx, pod, &client.DeleteOptions{GracePeriodSeconds: kptr.Int64(0)}); err != nil {
```

Comment:
Imagine the pods have some sort of finalizer on them. These delete calls can be asynchronous (even if we force). 

Instead, I think we should rely simply on len(pods) > 0. This will ensure that there are actually 0 pods when we delete the node. It has a side benefit of also deduping the boolean and continue statement.

Reply:
Since we wait for the gracePeriodSeconds here before actually force terminating, the pods will have been evicting + in the terminating state, and would have triggered finalizer workflows. Since we don't know how long finalizers will take, I think it's fair to say that after the gracePeriodSeconds we can attempt to force terminate. 

We need some force delete pod logic. If we don't, we will never have `len(pods) == 0` if the Kubelet is down and can't register the pod as notReady. Are you suggesting that these delete calls lie somewhere else?